reconcile/utils/terrascript_client.py

```diff
@@ -3892,12 +3917,72 @@ def populate_tf_resource_asg(self, resource, namespace_info):
                 {
                     "resource_type": "volume",
                     "tags": tags
-                }
-            ]
+                }]
         }
 
-        region = common_values.get('region') or \
-            self.default_regions.get(account)
+        image = common_values.get('image')
+        image_id = image.get('id')
+        if not image_id:
+            accounts = [a for a in self.filtered_accounts
+                        if a['name'] == account]
+            awsapi = AWSApi(1, accounts,
+                            settings=self.settings,
+                            init_users=False)
+            owner = image.get('owner')
+            if owner in self.uids.keys():
+                owner = self.uids.get(owner)
+
+            url = image.get('url')
+            ref = image.get('ref')
+            pattern = r'^[0-9a-f]{40}$'
+            upstream = image.get('upstream')
+            # get commit_sha from ref
+            if re.search(pattern, ref):
+                commit_sha = ref
+            # get commit_sha from upstream jenkins job
+            elif upstream:
+                instance_name = upstream['instance']['name']
+                token = upstream['instance']['token']
+                if instance_name not in self.jenkins_map:
+                    with self.jenkins_lock:
+                        if instance_name not in self.jenkins_map:
+                            self.jenkins_map[instance_name] = \
+                                JenkinsApi(token, ssl_verify=False,
+                                           settings=self.settings)
+                jenkins_instance = self.jenkins_map[instance_name]
+                job_name = upstream['name']
+                commit_sha = \
+                    jenkins_instance.get_last_success_build_commit(job_name)
+            # get commit_sha from branch
+            elif 'github' in url:
+                if not self.github:
+                    with self.github_lock:
+                        if not self.github:
```

Comment:
Why are you rechecking for `self.github` 2 lines after checking? Can multiple threads really work on this same object?

Reply:
I copied from https://github.com/app-sre/qontract-reconcile/blob/master/reconcile/utils/terrascript_client.py#L228-L231